plugins/hypervisors/vmware/src/com/cloud/storage/resource/VmwareStorageSubsystemCommandHandler.java

```diff
@@ -66,19 +68,28 @@ public VmwareStorageSubsystemCommandHandler(StorageProcessor processor, Integer
         this._nfsVersion = nfsVersion;
     }
 
-    /**
-     * Reconfigure NFS version for storage operations
-     * @param nfsVersion NFS version to set
-     * @return true if NFS version could be configured, false in other case
-     */
-    public boolean reconfigureNfsVersion(Integer nfsVersion){
+    public boolean reconfigureStorageProcessor(EnumMap<VmwareStorageProcessorConfigurableFields,Object> params) {
+        VmwareStorageProcessor processor = (VmwareStorageProcessor) this.processor;
         try {
-            VmwareStorageProcessor processor = (VmwareStorageProcessor) this.processor;
-            processor.setNfsVersion(nfsVersion);
-            this._nfsVersion = nfsVersion;
+            for (VmwareStorageProcessorConfigurableFields key : params.keySet()){
+                switch (key){
+                case NFS_VERSION:
+                    Integer nfsVersion = (Integer) params.get(key);
+                    processor.setNfsVersion(nfsVersion);
+                    this._nfsVersion = nfsVersion;
+                    break;
+                case FULL_CLONE_FLAG:
+                    boolean fullClone = (boolean) params.get(key);
+                    processor.setFullCloneFlag(fullClone);
+                    break;
+                default:
+                    String msg = "Unknown reconfigurable field " + key.getName() + " for VmwareStorageProcessor";
+                    throw new Exception(msg);
```

Comment:
General Java best practice never to throw ``Exception`` because it does not provide specific cause information.  Consider using ``IllegalStateException`` instead.